pandas/indexes/multi.py

```diff
@@ -2367,5 +2371,27 @@ def _sparsify(label_list, start=0, sentinel=''):
     return lzip(*result)
 
 
+def _check_equal_length(seq_of_seqs):
+    """
+    Ensure that all sequences in seq_of_seqs are the same length.
+
+    Since this function is time critical, it does zero error checking.
+    A TypeError will be raised if inner sequence does not support len().
+
+    Return True if all sequences are the same length, otherwise False
+    If seq_of_seqs is empty return True as well.
+    """
```

Comment:
do a 

```
if not is_list_like(seq_of_seqs):
     return True
```

at the top, this will prevent things like: strings & Timestamps from pass thru here (though they shouldn't be passed in the first place, its possible)